### PR TITLE
use new helpers and solidify arg logic

### LINKED ISSUE #1565: Spaces in parent folder file name bug
v5.7.1

### Describe the bug**
If there are spaces in the parent folder file name e.g. "07 Your first Fsharp Program" the Ionide build will fail, i.e. clicking on the green build arrow. However it works if pressing on the "bug" icon to build with debugging. Building and running at the CLI works find.

If there are no spaces e.g. "07_Your_first_Fsharp_Program" everything works fine.

### Screenshots

image

image

Running on Windows Surface Laptop 3 with latest Windows 11 preview build.

Same problem with Windows 10 however MacOS does not have the problem.

### PR DESCRIPTION
**Pull request by baronfel**
The companion to https://github.com/ionide/ionide-vscode-helpers/pull/43, and fixes #1565.

### CODE REVIEW

File: src/Core/Project.fs [hunk 1]
```diff
@@ -456,32 +456,38 @@ module Project =
...
             match dotnet with
             | Some dotnet ->
-                return Process.spawnWithShell dotnet "" cmd
+                return Process.spawnWithShell dotnet cmd
             | None -> return! Promise.reject "dotnet binary not found"
         }
 
     let private execWithShell exe cmd =
         promise {
-            return Process.spawnWithShell exe "mono" cmd
```
Comment (Krzysztof-Cieslak) [inline src/Core/Project.fs:479]: Mono 😱
Do we use this function anywhere?

Comment (baronfel) [reply src/Core/Project.fs:479]: dont think so, but i'll check and burn it with fire if we can.

Comment (baronfel) [reply src/Core/Project.fs:479]: oh I already had removed all of this :p

**Comment by baronfel** [comment]
closing, implicitly done via other PRs

**Pull request closed without merging**